collect_app/src/main/java/org/odk/collect/android/tasks/ImageLoadingTask.java

```diff
@@ -0,0 +1,93 @@
+package org.odk.collect.android.tasks;
+
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.provider.MediaStore;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.app.Fragment;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.exception.GDriveConnectionException;
+import org.odk.collect.android.fragments.dialogs.ProgressDialogFragment;
+import org.odk.collect.android.utilities.FileUtils;
+import org.odk.collect.android.utilities.ImageConverter;
+import org.odk.collect.android.utilities.MediaUtils;
+import org.odk.collect.android.utilities.ToastUtils;
+
+import java.io.File;
+import java.lang.ref.WeakReference;
+
+import timber.log.Timber;
+
+public class ImageLoadingTask extends AsyncTask<Uri, Void, File> {
+
+    private WeakReference<FormEntryActivity> formEntryActivity;
+
+    public ImageLoadingTask(FormEntryActivity formEntryActivity) {
+        onAttach(formEntryActivity);
+    }
+
+    public void onAttach(FormEntryActivity formEntryActivity) {
+        this.formEntryActivity = new WeakReference<>(formEntryActivity);
+    }
+
+    public void onDetach() {
+        this.formEntryActivity = null;
+    }
+
+    @Override
+    protected File doInBackground(Uri... uris) {
+
+        File instanceFile = null;
+
+        if (formEntryActivity.get().getFormController() != null) {
+            instanceFile = formEntryActivity.get().getFormController().getInstanceFile();
```

Comment:
it's better to create a FormController object and use it in both places than getting it from formEntryActivity twice.